apis/accounts/index.yaml

```diff
@@ -0,0 +1,148 @@
+                example: "150"
+              total_size:
+                type: number
+                description: "Total count matching the request."
+              accounts:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    validating_pubkey:
+                      type: string
+                      description: "The validating public key."
+                    derivation_path:
+                      type: string
+                      description: "The derivation path (if present in the imported keystore)."
+                    slashing_protection_last_entry:
+                      type: string
+                      description: |
+                        Latest slashing protection DB data item associated with this pubkey.
+                        Signal no slashing protection DB data with a value of TBD (0, null, "")
+
+    "401":
+      $ref: "../../validator-oapi.yaml#/components/responses/Unauthorized"
+    "403":
+      $ref: "../../validator-oapi.yaml#/components/responses/Forbidden"
+    "500":
+      $ref: "../../validator-oapi.yaml#/components/responses/InternalError"
+
+post:
+  tags:
+    - Accounts
+  operationId: "ImportAccounts"
+  summary: "Import accounts."
+  description: |
+    Import keystores generated by the Eth2.0 deposit CLI tooling. All keystores MUST be encrypted with the same password.
+  security:
+    - cookieAuth: []
+  requestBody:
+    content:
+      application/json:
+        schema:
+          type: object
+          properties:
+            keystores_imported:
+              type: array
+              description: "JSON-encoded keystore files to import during wallet creation."
+              items:
+                type: object
+            keystores_password:
+              description: "Password to unlock imported keystore files."
+              type: string
+  responses:
+    "200":
+      description: Success response
+      content:
+        application/json:
+          schema:
+            title: ImportAccountsResponse
+            type: object
+            properties:
+              imported_pubkeys:
+                type: array
+                description: "List of successfully imported pubkeys."
+                items:
+                  type: string
+    "401":
+      $ref: "../../validator-oapi.yaml#/components/responses/Unauthorized"
+    "403":
+      $ref: "../../validator-oapi.yaml#/components/responses/Forbidden"
+    "500":
+      $ref: "../../validator-oapi.yaml#/components/responses/InternalError"
+
+delete:
+  tags:
+    - Accounts
+  operationId: "DeleteAccounts"
+  summary: "Delete accounts."
+  description: |
+    Delete keystores for the pubkeys provided in the request. The validator client MUST stop validating
+    with these keystores immediately and MUST return a 200 status only after confirming no more signatures
+    can be created with those keys.
+  security:
```

Comment:
This operation is unsafe one. Without proper authentication its very easy for somebody drop all the accounts from validator's client database. The only thing which you should know is validator's public key. I think even cookie (and mostly because of cookie authentication) is not enough here. I think the most appropriate one is to request `private key` or `keystore + password` to confirm deletion. So if you key owner it will be not a problem for you, but if you some stranger you obviously could not recover validator's private key from known public key.

Reply:
Sending the `private key` or `keystore + password` seems like a much worse trade-off to achieve authentication.

Reply:
You already sending this pair while adding key, so this is not a trade-off. If you want to avoid it, you need to create some kind of token when adding keystores, so this tokens will not be known by public.

Reply:
Using the method @paulhauner described at the `operationId: "Login"` using api keys (secrets) would resolve this.